src/main/java/com/redhat/labs/lodestar/service/EngagementService.java

```diff
@@ -199,6 +201,65 @@ void validateCustomerAndProjectNames(Engagement toUpdate, Engagement existing) {
 
     }
 
+    /**
+     * Throws {@link WebApplicationException} if the supplied subdomain is used by
+     * another {@link Engagement}.
+     * 
+     * @param engagement
+     */
+    void validateSubdomainOnCreate(Engagement engagement) {
+
+        String subdomain = engagement.getOcpSubDomain();
+        if (doesSubdomainExist(engagement.getOcpSubDomain())) {
+            throw new WebApplicationException(
+                    String.format("The requested subdomain, %s, is already in use", subdomain), HttpStatus.SC_CONFLICT);
+        }
+
+    }
+
+    /**
+     * Throws {@link WebApplicationException} if the supplied subdomain is different
+     * from the persisted domain and another {@link Engagement} is already using it.
+     * 
+     * @param toUpdate
+     * @param existing
+     */
+    void validateSubdomainOnUpdate(Engagement toUpdate, Engagement existing) {
+
+        String subdomain = toUpdate.getOcpSubDomain();
+        if (subdomain != null && !subdomain.equalsIgnoreCase(existing.getOcpSubDomain())
+                && doesSubdomainExist(subdomain)) {
+            throw new WebApplicationException(
+                    String.format("The requested subdomain, %s, is already in use", subdomain), HttpStatus.SC_CONFLICT);
+        }
+
+    }
+
+    /**
+     * Return false if subdomain is null, empty, or is not found in the data store.  Otherwise, true.
+     * 
+     * @param subdomain
+     */
+    boolean doesSubdomainExist(String subdomain) {
+
+        if (subdomain == null || subdomain.isBlank() || getBySubdomain(subdomain).isEmpty()) {
```

Comment:
not a big fan of `return true` how do you feel about removing the if statement and just have a `return (equality test)`

Reply:
updated these two methods as there were no null/blank checks on the direct api call.  used the suggested equality test in stead of if statement.